### PR TITLE
cmd/root: randomize pivot container name

### PR DESCRIPTION
**Pull request by jlebon**
This is a backport of
https://github.com/openshift/machine-config-operator/pull/1285, which
for the 4.1 release should live here since we were still using the
separate pivot package.

Resolves: https://bugzilla.redhat.com/show_bug.cgi?id=1783621

Original cherry-picked commit message:

If podman has a bug and cannot cleanup container names, we'll be stuck
in the update process as seen in
https://bugzilla.redhat.com/show_bug.cgi?id=1775647.
Randomize container names with UUID to avoid that and I think it's
generally better to do this w/o cleaning up at the beginning every time.

Signed-off-by: Antonio Murdaca <runcom@linux.com>

### CODE REVIEW

**Comment by jlebon** [comment]
Alternatively... move 4.1 to the MCD-integrated pivot service? There is zero CI here so this is much riskier.

**Comment by jlebon** [comment]
https://github.com/openshift/machine-config-operator/pull/1285#issuecomment-567196188

File: cmd/root.go [hunk 1]
```diff
@@ -307,11 +308,15 @@ func pullAndRebase(container string) (imgid string, changed bool) {
 		imgid = container
 	}
 
-	// Clean up a previous container
-	podmanRemove(types.PivotName)
```
Comment (cgwalters) [inline cmd/root.go]: We might as well keep this line to ensure we GC containers with the previous naming.

**Review by cgwalters — APPROVED** [review]

**Comment by cgwalters** [comment]
> Alternatively... move 4.1 to the MCD-integrated pivot service? There is zero CI here so this is much riskier.

Not trivial...bigger picture we want 4.1 users to get on the 4.2 train soon, and then soon 4.3...

**Comment by jlebon** [comment]
/hold
Still need to test this.

File: cmd/root.go (hunk 1 above)
Comment (jlebon) [reply cmd/root.go]: Done! (Though note the original patch didn't do this).

**Comment by jlebon** [comment]
/hold cancel

OK, tested this works now!

**Review by ashcrow — APPROVED** [review]
Backport looks good.

**Comment by ashcrow** [comment]
Backport the backport to https://github.com/openshift/pivot/commits/release-4.1?

**Pull request merged**